llama_stack/providers/adapters/memory/weaviate/weaviate.py

```diff
@@ -0,0 +1,196 @@
+from typing import List, Optional, Dict, Any
+from numpy.typing import NDArray
+
+import weaviate
+import weaviate.classes as wvc
+from weaviate.classes.init import Auth
+
+from llama_stack.apis.memory import *
+from llama_stack.distribution.request_headers import get_request_provider_data
+from llama_stack.providers.utils.memory.vector_store import (
+    BankWithIndex,
+    EmbeddingIndex,
+)
+
+from .config import WeaviateConfig, WeaviateRequestProviderData
+
+class WeaviateIndex(EmbeddingIndex):
+    def __init__(self, client: weaviate.Client, collection: str):
+        self.client = client
+        self.collection = collection
+
+    async def add_chunks(self, chunks: List[Chunk], embeddings: NDArray):
+        assert len(chunks) == len(embeddings), f"Chunk length {len(chunks)} does not match embedding length {len(embeddings)}"
+
+        data_objects = []
+        for i, chunk in enumerate(chunks):
+            print(f"Adding chunk #{i} tokens={chunk.token_count}")
+            
+            data_objects.append(wvc.data.DataObject(
+                properties={
+                    "chunk_content": chunk,
+                },
+                vector = embeddings[i].tolist()
+            ))
+
+        # Inserting chunks into a prespecified Weaviate collection
+        assert self.collection is not None, "Collection name must be specified"
+        my_collection = self.client.collections.get(self.collection)
+        
+        await my_collection.data.insert_many(data_objects)
+
+
+    async def query(self, embedding: NDArray, k: int) -> QueryDocumentsResponse:
+        assert self.collection is not None, "Collection name must be specified"
+
+        my_collection = self.client.collections.get(self.collection)
+        
+        results = my_collection.query.near_vector(
+            near_vector = embedding.tolist(),
+            limit = k,
+            return_meta_data = wvc.query.MetadataQuery(distance=True)
+        )
+
+        chunks = []
+        scores = []
+        for doc in results.objects:
+            try:
+                chunk = doc.properties["chunk_content"]
+                chunks.append(chunk)
+                scores.append(1.0 / doc.metadata.distance)
+            
+            except Exception as e:
+                import traceback
+                traceback.print_exc()
+                print(f"Failed to parse document: {e}")
+
+        return QueryDocumentsResponse(chunks=chunks, scores=scores)
+
+
+class WeaviateMemoryAdapter(Memory):
+    def __init__(self, config: WeaviateConfig) -> None:
+        self.config = config
+        self.client = None
+        self.cache = {}
+
+    def _get_client(self) -> weaviate.Client:
+            request_provider_data = get_request_provider_data()
+            
+            if request_provider_data is not None:
+                assert isinstance(request_provider_data, WeaviateRequestProviderData)
+
+                print(f"WEAVIATE API KEY: {request_provider_data.weaviate_api_key}")
```

Comment:
do we really want to print the api key into the console?

Reply:
removed